src/osp/Active/CommonPhysics.h

```diff
@@ -0,0 +1,135 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "../types.h"
+
+namespace osp::phys
+{
+
+// Formerly PrototypePart
+enum class ECollisionShape : uint8_t
+{
+    NONE,
+    COMBINED,
+    SPHERE,
+    BOX,
+    CAPSULE,
+    CYLINDER,
+    //MESH,
+    CONVEX_HULL,
+    TERRAIN
+};
+
+float col_shape_volume(ECollisionShape shape, Vector3 scale);
+
+// Formerly SysNewton
+/**
+ * Generic rigid body state
+ */
+struct DataRigidBody
+{
+    // Modify these
+    Vector3 m_inertia{1, 1, 1};
+    Vector3 m_netForce{0, 0, 0};
+    Vector3 m_netTorque{0, 0, 0};
+
+    float m_mass{1.0f};
+    Vector3 m_velocity{0, 0, 0};
+    Vector3 m_rotVelocity{0, 0, 0};
+    Vector3 m_centerOfMassOffset{0, 0, 0};
+
+    bool m_colliderDirty{false}; // set true if collider is modified
+};
+
+/**
+ * Transform an inertia tensor
+ *
+ * Transforms an inertia tensor using the parallel axis theorem.
+ * See "Tensor generalization" section on
+ * https://en.wikipedia.org/wiki/Parallel_axis_theorem for more information.
+ *
+ * @param I           [in] The original inertia tensor
+ * @param mass        [in] The total mass of the object
+ * @param translation [in] The translation part of the transformation
+ * @param rotation    [in] The rotation part of the transformation
+ *
+ * @return The transformed inertia tensor
+ */
+Matrix3 transform_inertia_tensor(Matrix3 I, float mass,
+    Vector3 translation, Matrix3 rotation);
+
+/**
+ * Compute the inertia tensor for a collider shape
+ * 
+ * Automatically selects the correct function necessary to compute the inertia
+ * for the given shape
+ * 
+ * @param shape [in] The shape of the collider
+ * @param scale [in] The (x, y, z) scale of the collider
+ * @param mass  [in] The total mass of the collider
+ * 
+ * @return The moment of inertia about the principle axes (x, y, z)
+ */
+Vector3 collider_inertia_tensor(ECollisionShape shape, Vector3 scale, float mass);
+
+/**
+ * Compute the inertia tensor for a cylinder
+ *
+ * Computes the moment of inertia about the principal axes of a cylinder
+ * with specified mass, height, and radius, whose axis of symmetry lies
+ * along the z-axis
+ *
+ * @return The moment of inertia about the 3 principal axes (x, y, z)
+ */
+constexpr Vector3 cylinder_inertia_tensor(float radius, float height, float mass);
```

Comment:
Magnum doesn't have any of these?

Might be worth a PR to them.

Reply:
Magnum doesn't really concern itself with physics stuff, do you mean Newton? It may, but it probably won't for all the shapes we need, and I just prefer to do it myself.

Reply:
So is this a Magnum::Vector3 ?

Or is this a Newton::Vector3 ?

Maybe not all of it is something Magnum would care about, but i suspect at least some of these operations would be available, or something that Magnum would upstream from us.

Reply:
Magnum is generally physics-agnostic since it's just a graphics layer, it just has basic math types like vectors and matrices because they're relevant to graphics. We can try but I don't think it's relevant to its purpose.